airbyte-server/src/main/java/io/airbyte/server/handlers/WebBackendConnectionsHandler.java

```diff
@@ -654,6 +654,11 @@ protected static ConnectionUpdate toConnectionPatch(final WebBackendConnectionUp
                                                       final List<UUID> finalOperationIds) {
     final ConnectionUpdate connectionPatch = new ConnectionUpdate();
 
+    if (webBackendConnectionPatch.getSyncCatalog() != null) {
+      // the user has updated their sync catalog, so there is no breaking change
```

Comment:
Should we check that all the change are now not breaking?

Reply:
this was considered in the tech design but it requires getting the source schema and calculating a diff again. what do you think?

Reply:
this is a simpler way of doing it based on FE validations (they can't update an invalid sync catalog), but maybe it's better for us to re-validate here too

Reply:
just realized the env variable won't work if we do this, since we're marking connections as broken even in the absence of the env variable. so I'll have to update to fetch the most recent actor catalog and do a diff to check whether it's broken